c_level_db.go

```diff
@@ -50,79 +51,95 @@ func NewCLevelDB(name string, dir string) (*CLevelDB, error) {
 }
 
 // Implements DB.
-func (db *CLevelDB) Get(key []byte) []byte {
+func (db *CLevelDB) Get(key []byte) ([]byte, error) {
 	key = nonNilBytes(key)
 	res, err := db.db.Get(db.ro, key)
 	if err != nil {
-		panic(err)
+		return nil, err
 	}
-	return res
+	return res, nil
 }
 
 // Implements DB.
-func (db *CLevelDB) Has(key []byte) bool {
-	return db.Get(key) != nil
+func (db *CLevelDB) Has(key []byte) (bool, error) {
+	_, err := db.Get(key)
+	if err != nil {
+		return false, err
+	}
+	return true, nil
 }
 
 // Implements DB.
-func (db *CLevelDB) Set(key []byte, value []byte) {
+func (db *CLevelDB) Set(key []byte, value []byte) error {
 	key = nonNilBytes(key)
 	value = nonNilBytes(value)
-	err := db.db.Put(db.wo, key, value)
-	if err != nil {
-		panic(err)
+	if err := db.db.Put(db.wo, key, value); err != nil {
+		return err
 	}
+	return nil
 }
 
 // Implements DB.
-func (db *CLevelDB) SetSync(key []byte, value []byte) {
+func (db *CLevelDB) SetSync(key []byte, value []byte) error {
 	key = nonNilBytes(key)
 	value = nonNilBytes(value)
-	err := db.db.Put(db.woSync, key, value)
-	if err != nil {
-		panic(err)
+	if err := db.db.Put(db.woSync, key, value); err != nil {
+		return err
 	}
+	return nil
 }
 
 // Implements DB.
-func (db *CLevelDB) Delete(key []byte) {
+func (db *CLevelDB) Delete(key []byte) error {
 	key = nonNilBytes(key)
-	err := db.db.Delete(db.wo, key)
-	if err != nil {
-		panic(err)
+	if err := db.db.Delete(db.wo, key); err != nil {
+		return err
 	}
+	return nil
 }
 
 // Implements DB.
-func (db *CLevelDB) DeleteSync(key []byte) {
+func (db *CLevelDB) DeleteSync(key []byte) error {
 	key = nonNilBytes(key)
-	err := db.db.Delete(db.woSync, key)
-	if err != nil {
-		panic(err)
+	if err := db.db.Delete(db.woSync, key); err != nil {
+		return err
 	}
+	return nil
 }
 
 func (db *CLevelDB) DB() *levigo.DB {
 	return db.db
 }
 
 // Implements DB.
-func (db *CLevelDB) Close() {
+func (db *CLevelDB) Close() error {
```

Comment:
not sure if returning an error here would benefit anybody really

Reply:
I was thinking the same but on some impl the close returns an error, boltdb close func returns an error so I didnt want to hide it